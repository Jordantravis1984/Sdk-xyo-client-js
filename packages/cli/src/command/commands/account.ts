import { parse } from 'path'
import { ArgumentsCamelCase, Argv, CommandBuilder, CommandModule } from 'yargs'

import { opts } from '../requireDirectoryOptions'
// eslint-disable-next-line @typescript-eslint/ban-types
type Arguments = {}

export const aliases: ReadonlyArray<string> = []
export const builder: CommandBuilder = (yargs: Argv) =>
  yargs.usage('Usage: $0 account <command> [Options]').commandDir(parse(__filename).name, opts).demandCommand().version(false)
export const command = 'account <command> [Options]'
export const deprecated = false
export const describe = 'Create & manage your XYO account'
export const handler = function (_argv: ArgumentsCamelCase<Arguments>) {
  // do something with argv.
  console.log('handler')
}

const mod: CommandModule = {
  aliases,
  command,
  deprecated,
  describe,
  handler,
}

// eslint-disable-next-line import/no-default-export
export default mod
