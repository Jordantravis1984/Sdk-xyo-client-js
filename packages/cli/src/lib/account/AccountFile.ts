export interface AccountFile {
  mnemonic: string
}
