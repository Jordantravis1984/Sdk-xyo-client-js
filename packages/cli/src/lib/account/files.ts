import { homedir } from 'os'
import { join } from 'path'

const fileName = '.xyo.seed'

/**
 * The file path where the Account information is stored
 */
export const accountFile = join(homedir(), fileName)
