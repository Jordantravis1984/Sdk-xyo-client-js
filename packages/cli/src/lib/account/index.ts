export * from './files'
export * from './getAccount'
export * from './getAccountInfo'
