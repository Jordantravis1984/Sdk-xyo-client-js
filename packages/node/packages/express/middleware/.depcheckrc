ignores: [
  "@xyo-network/express-node-test"
]