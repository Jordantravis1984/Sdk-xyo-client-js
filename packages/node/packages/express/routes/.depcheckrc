ignores: [
  "@xyo-network/express-node-test",
  "@xylabs/ts-scripts-yarn3",
  "tslib"
]