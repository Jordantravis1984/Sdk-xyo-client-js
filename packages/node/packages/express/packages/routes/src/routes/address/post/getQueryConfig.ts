import { XyoBoundWitness, XyoBoundWitnessSchema } from '@xyo-network/boundwitness-model'
import { requestCanAccessArchive } from '@xyo-network/express-node-lib'
import { AbstractModule, Module, ModuleConfig, ModuleConfigSchema, XyoQueryBoundWitness } from '@xyo-network/module'
import { ArchiveModuleConfig } from '@xyo-network/node-core-model'
import { XyoPayload } from '@xyo-network/payload-model'
import { Request } from 'express'

export const getQueryConfig = async (
  mod: Module,
  req: Request,
  bw: XyoQueryBoundWitness,
  payloads?: XyoPayload[],
): Promise<ModuleConfig | undefined> => {
  const archivist = mod as unknown as AbstractModule
  const config = archivist?.config as unknown as ArchiveModuleConfig
  const archive = config?.archive
  if (archive && (await requestCanAccessArchive(req, archive))) {
    // Recurse through payloads for nested BWs
    const nestedBwAddresses =
      payloads
        ?.flat(5)
        .filter<XyoBoundWitness>((payload): payload is XyoBoundWitness => payload?.schema === XyoBoundWitnessSchema)
        .map((bw) => bw.addresses) || []
    const addresses = [bw.addresses, ...nestedBwAddresses].filter((address) => address.length)
    const allowed = addresses.length ? Object.fromEntries(archivist.queries.map((schema) => [schema, addresses])) : {}
    const security = { allowed }
    return { schema: ModuleConfigSchema, security }
  }
}
