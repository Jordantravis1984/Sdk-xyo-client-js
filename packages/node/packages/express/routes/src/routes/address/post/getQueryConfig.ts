import { XyoBoundWitness, XyoBoundWitnessSchema } from '@xyo-network/boundwitness-model'
import { requestCanAccessArchive } from '@xyo-network/express-node-lib'
import { AbstractModule, AbstractModuleConfig, AbstractModuleConfigSchema, Module, XyoQueryBoundWitness } from '@xyo-network/module'
import { ArchiveModuleConfig } from '@xyo-network/node-core-model'
import { XyoPayload } from '@xyo-network/payload-model'
import { Request } from 'express'

export const getQueryConfig = async (
  mod: Module,
  req: Request,
  bw: XyoQueryBoundWitness,
  payloads?: XyoPayload[],
): Promise<AbstractModuleConfig | undefined> => {
  const archivist = mod as unknown as AbstractModule
  const config = archivist?.config as unknown as ArchiveModuleConfig
  const archive = config?.archive
  if (archive && (await requestCanAccessArchive(req, archive))) {
    // Recurse through payloads for nested BWs
    const nestedBwAddresses =
      payloads
        ?.flat(5)
        .filter<XyoBoundWitness>((payload): payload is XyoBoundWitness => payload?.schema === XyoBoundWitnessSchema)
        .map((bw) => bw.addresses) || []
    const allowed = Object.fromEntries(archivist.queries().map((schema) => [schema, [bw.addresses, ...nestedBwAddresses]]))
    const security = { allowed }
    return { schema: AbstractModuleConfigSchema, security }
  }
}
