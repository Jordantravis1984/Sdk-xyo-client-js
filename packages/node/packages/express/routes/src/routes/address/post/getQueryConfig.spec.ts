import { Account } from '@xyo-network/account'
import {
  AbstractModule,
  AbstractModuleConfig,
  AbstractModuleConfigSchema,
  AbstractModuleDiscoverQuerySchema,
  QueryBoundWitnessBuilder,
} from '@xyo-network/module'
import { AbstractArchivist } from '@xyo-network/modules'
import { ArchiveModuleConfig, ArchiveModuleConfigSchema } from '@xyo-network/node-core-model'
import { Request } from 'express'
import { mock } from 'jest-mock-extended'

import { getQueryConfig } from './getQueryConfig'

let canAccess = true
jest.mock('@xyo-network/express-node-lib', () => ({
  requestCanAccessArchive: jest.fn(() => Promise.resolve(canAccess)),
}))
const req = mock<Request>()
const testAccount1 = new Account({ phrase: 'testPhrase1' })
const testAccount2 = new Account({ phrase: 'testPhrase2' })

describe('getQueryConfig', () => {
  describe('with module', () => {
    const mod = mock<AbstractModule>()
    const config: AbstractModuleConfig = { schema: AbstractModuleConfigSchema }
    mod.config = config
    it('returns undefined', async () => {
      const query = new QueryBoundWitnessBuilder().query({ schema: AbstractModuleDiscoverQuerySchema }).build()
      const config = await getQueryConfig(mod, req, query[0], query[1])
      expect(config).toBeUndefined()
    })
  })
  describe('with archivist', () => {
    const archive = 'temp'
    const config: ArchiveModuleConfig = { archive, schema: ArchiveModuleConfigSchema }
    const queries = () => [AbstractModuleDiscoverQuerySchema]
    const mod = mock<AbstractArchivist>({
      config,
      queries,
    })
    describe('when request can access archive', () => {
      beforeAll(() => {
        canAccess = true
      })
      it('generates config for single-signer requests', async () => {
        const query = new QueryBoundWitnessBuilder().query({ schema: AbstractModuleDiscoverQuerySchema }).witness(testAccount1).build()
        const config = await getQueryConfig(mod, req, query[0], query[1])
        expect(config).toMatchSnapshot()
      })
      it('generates config for multi-signer requests', async () => {
        const query = new QueryBoundWitnessBuilder()
          .query({ schema: AbstractModuleDiscoverQuerySchema })
          .witness(testAccount1)
          .witness(testAccount2)
          .build()
        const config = await getQueryConfig(mod, req, query[0], query[1])
        expect(config).toMatchSnapshot()
      })
      it('generates config for unsigned requests', async () => {
        const query = new QueryBoundWitnessBuilder().query({ schema: AbstractModuleDiscoverQuerySchema }).build()
        const config = await getQueryConfig(mod, req, query[0], query[1])
        expect(config).toMatchSnapshot()
      })
    })
    describe('when request cannot access archive', () => {
      beforeAll(() => {
        canAccess = false
      })
      it('returns undefined', async () => {
        const query = new QueryBoundWitnessBuilder().query({ schema: AbstractModuleDiscoverQuerySchema }).witness(testAccount1).build()
        const config = await getQueryConfig(mod, req, query[0], query[1])
        expect(config).toBeUndefined()
      })
    })
  })
})
