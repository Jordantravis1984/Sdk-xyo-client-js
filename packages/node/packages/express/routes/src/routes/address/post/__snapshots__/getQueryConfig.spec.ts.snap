// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`getQueryConfig with archivist when request can access archive generates config for multi-signer requests 1`] = `
{
  "schema": "network.xyo.module.config",
  "security": {
    "allowed": {
      "network.xyo.query.module.discover": [
        [
          "3a15a6c96665931b76c1d2a587ea1132dbfdc266",
          "b3bbeb946d88cd5fe70146a6bb276c7b384f6582",
        ],
      ],
    },
  },
}
`;

exports[`getQueryConfig with archivist when request can access archive generates config for signed requests 1`] = `
{
  "schema": "network.xyo.module.config",
  "security": {
    "allowed": {
      "network.xyo.query.module.discover": [
        [
          "3a15a6c96665931b76c1d2a587ea1132dbfdc266",
        ],
      ],
    },
  },
}
`;

exports[`getQueryConfig with archivist when request can access archive generates config for single-signer requests 1`] = `
{
  "schema": "network.xyo.module.config",
  "security": {
    "allowed": {
      "network.xyo.query.module.discover": [
        [
          "3a15a6c96665931b76c1d2a587ea1132dbfdc266",
        ],
      ],
    },
  },
}
`;

exports[`getQueryConfig with archivist when request can access archive generates config for unsigned requests 1`] = `
{
  "schema": "network.xyo.module.config",
  "security": {
    "allowed": {},
  },
}
`;
